src/components/main/payment/PaymentValidation.ts

```diff
@@ -0,0 +1,244 @@
+        }
+      }
+
+      if (cardValue.length < 16) {
+        return false;
+      }
+
+      return true;
+    });
+
+    PaymentValidation.check(cardValid, Validation.CardValidInvalidMessage, (): boolean => {
+      if (cardValid.value.length == 2) {
+        if (this.keepOnlyDigits(cardValid.value).length == 2) {
+          cardValid.value = this.keepOnlyDigits(cardValid.value) + '/';
+        }
+        return false;
+      }
+
+      const value = this.keepOnlyDigits(cardValid.value);
+
+      if (cardValid.value.length - value.length != 1) {
+        return false;
+      }
+
+      const cardValue = cardValid.value.trim();
+      const splitedValue = cardValue.split('/');
+      const monthValue = splitedValue[0];
+      const yearValue = splitedValue[1];
+
+      if (Number(monthValue) > 12 || Number(monthValue) < 1 || Number(yearValue) < 23) {
+        return false;
+      }
+
+      return true;
+    });
+
+    PaymentValidation.check(cardCvv, Validation.CardCvvInvalidMessage, (): boolean => {
+      cardCvv.value = this.keepOnlyDigits(cardCvv.value);
+      return cardCvv.value.length === 3;
+    });
+
+    form.addEventListener('submit', (event) => {
+      const invalidInputs = inputFormArr.filter((item) => {
+        if (!item.parentElement) {
+          return false;
+        }
+
+        const error = item.parentElement.querySelector('.form__input_error');
+
+        if (!error) {
+          return false;
+        }
+
+        return error.classList.contains('form__input_error_active');
+      });
+
+      if (invalidInputs.length > 0) {
+        invalidInputs.forEach((input: HTMLElement) =>
+          PaymentValidation.showMessageError(input, Validation.ErrorMessage),
+        );
+      } else {
+        const bodyContainer = document.querySelector('.body');
+        let seconds = 5;
+
+        const popup = document.querySelector('.popup');
+        const popupOverlay = document.querySelector('.popup__overlay');
+
+        if (popup) {
+          popup.remove();
+        }
+
+        if (bodyContainer && popupOverlay && bodyContainer) {
+          const closePopup = document.createElement('div');
+          closePopup.classList.add('popup_close');
+          closePopup.textContent = `Thank you for your order. Redirect to the store after ${seconds} sec.`;
+          bodyContainer.append(closePopup);
+
+          const showText = setInterval(() => {
+            seconds -= 1;
+            closePopup.textContent = `Thanks for your order. Redirect to the store after ${seconds} sec.`;
+          }, 1000);
+
+          setTimeout(() => {
+            closePopup.remove();
+            popupOverlay.remove();
+            bodyContainer.classList.remove('body_active');
+            clearInterval(showText);
+            LocalStorage.removeProductsFromCart();
+            window.location.hash = `#${Pages.Main}`;
+          }, 5000);
+        }
+      }
+      event.preventDefault();
+    });
```

Comment:
Too much logic in the event listener callback. Split it into multiple methods as `validateSmth` & `createRedirectionTimer`